src/hooks/useArticleTextSize.js

```diff
@@ -1,12 +1,18 @@
-import { useState, useLayoutEffect } from 'preact/hooks'
+import { useState, useEffect } from 'preact/hooks'
 import { articleTextSize } from 'utils'
 
-export const useArticleTextSize = (dependencies = []) => {
-  const [textSize, setTextSize] = useState(articleTextSize.get())
+let setArticleFontClass
 
-  useLayoutEffect(() => {
-    articleTextSize.init()
-  }, dependencies)
+export const useArticleTextSize = (origin) => {
+  const [fontSizeClass, setFontSizeClass] = useState(articleTextSize.getFontSizeClassName())
 
-  return [textSize, setTextSize]
+  if (origin === 'Article') {
+    setArticleFontClass = setFontSizeClass
+  }
+
+  useEffect(() => {

```

Comment:
`useEffect` without dependencies will run every time any component using this hook renders. It seems strange that the font size class is set on every render. Shouldn't it be set only when it changes in the UI with the slider?